### ISSUE TITLE
How to get forward method of model

### ISSUE THREAD

**Issue by VijayRajIITP** [ollama/ollama#7211]
### What is the issue?

I am using ollama
ollama pull llama3
llm=Ollama(model='llama3')
base_url="http://localhost:11434/v1"    
l want to work with foraward method of model how to get that .Is it poosible

### OS

Linux

### GPU

Nvidia

**Comment by rick-github** [comment]
What is `forward method`?

**Comment by VijayRajIITP** [comment]
function that computes the output of the model given an input. It encapsulates the entire forward pass through the layers of the model, processing the input data and producing predictions or embeddings.
Encoding and decoding

**Comment by rick-github** [comment]
Are you referring to logprobs?  https://github.com/ollama/ollama/issues/2415

**Comment by VijayRajIITP** [comment]
from langchain.from langchain.llms import Ollama
llm=Ollama(model='llama3')
The response going through rest api using post method , can we acess server side code as it is local host , I want to see the internal working how my input is going to lma model and all the internal detail of model in .py itself

**Comment by rick-github** [comment]
You can use `strace` to see the internal detail:
```
strace python3 my_ollama_script.py
```

**Comment by VijayRajIITP** [comment]
I’m trying to understand how Ollama manages tokenization and detokenization when interacting with APIs. I used debug tools to inspect the process, but the responses received from the API are still in text form. I’m interested in the internal workings that occur after a response is processed through the API, such as how tokenization and detokenization are handled.

I also explored the Ollama installation directory, which I set up using the command:

```bash
curl -fsSL https://ollama.com/install.sh | sh
```

However, I couldn’t find any files related to the model there. This raises the question of how the model is loaded, how it processes the input, and how the output is post-processed.
